diff --git a/src/FFmpegWriter.cpp b/src/FFmpegWriter.cpp
index c183b86..1aea034 100644
--- a/src/FFmpegWriter.cpp
+++ b/src/FFmpegWriter.cpp
@@ -1515,0 +1516,3 @@ void FFmpegWriter::open_video(AVFormatContext *oc, AVStream *st) {
+		if (info.vcodec.find("_nvenc") != std::string::npos) {
+			av_opt_set(video_codec_ctx->priv_data, "rc", "vbr", 0);
+		}
@@ -1520,4 +1523,10 @@ void FFmpegWriter::open_video(AVFormatContext *oc, AVStream *st) {
-				video_codec_ctx->profile = AV_PROFILE_H264_CONSTRAINED_BASELINE;
-				av_opt_set(video_codec_ctx->priv_data, "preset", "slow", 0);
-				av_opt_set(video_codec_ctx->priv_data, "tune", "zerolatency", 0);
-				av_opt_set(video_codec_ctx->priv_data, "vprofile", "baseline", AV_OPT_SEARCH_CHILDREN);
+				if (info.vcodec.find("_nvenc") != std::string::npos) {
+					video_codec_ctx->profile = AV_PROFILE_H264_HIGH;
+					av_opt_set(video_codec_ctx->priv_data, "preset", "p4", 0);
+					av_opt_set(video_codec_ctx->priv_data, "tune", "hq", 0);
+				} else {
+					video_codec_ctx->profile = AV_PROFILE_H264_CONSTRAINED_BASELINE;
+					av_opt_set(video_codec_ctx->priv_data, "preset", "slow", 0);
+					av_opt_set(video_codec_ctx->priv_data, "tune", "zerolatency", 0);
+					av_opt_set(video_codec_ctx->priv_data, "vprofile", "baseline", AV_OPT_SEARCH_CHILDREN);
+				}
